app/Http/Controllers/Widgets/WidgetController.php

```diff
@@ -63,11 +63,17 @@ public function __invoke(Request $request)
 
         if ($this->show_settings) {
             $view = $this->getSettingsView($request);
-        } else {
-            $view = $this->getView($request);
         }
 
         $settings = $this->getSettings();
```

Comment:
If you move `$settings = $this->getSettings();` up, you don't need two if statements.

Reply:
tried it, but it will crash Widget. This is the only "elegant" way it works.